app/models/manageiq/providers/kubernetes/container_manager/scanning/job.rb

```diff
@@ -46,6 +46,7 @@ def start
     image = target_entity
     return queue_signal(:abort_job, "no image found", "error") unless image
     return queue_signal(:abort_job, "cannot analyze non docker images", "error") unless image.docker_id
+    # return queue_signal(:abort_job, "role-missing", "error") unless CURRENT_SERVER.active_roles.include?("smartproxy")
```

Comment:
At the time this code is executed, the request has already been taken off the queue - is that right?
If that's the case, then we would never get here if the "smartproxy" role isn't set - only appliances with the role should take those jobs off the queue.

I don't think its enough to check the current server, you'll have to check to see if there are any servers that have the role, and only fail if no server does. However, I think that check needs to be made before the job is queued. Even then, I'm not sure if that's a good thing to do. You may want to issue a warning and queue the job anyway, given the role may be enabled after the job is queued. I often use roles as switches to delay/initiate processes.